### DISCUSSION TITLE
Retrieve list of cryptocurrency symbols on yahoo finance

### DISCUSSION THREAD

**Discussion by fja05680** [dpguthrie/yahooquery#74] [General]
Hello,

Is there a way to retrieve the complete list of symbols on this page?

https://finance.yahoo.com/cryptocurrencies

thanks, great library!

Farrell

**Comment by dpguthrie** [comment]
Have you looked at the `Screener` class?  It's not exactly that page, but there are some predefined screeners you can use that might get you all that data.

```
from yahooquery import Screener

s = Screener()
data = s.get_screeners('all_cryptocurrencies_us', count=250)

# data is in the quotes key
data['all_cryptocurrencies_us']['quotes']
```

Other crypto screeners you could use are:  `all_cryptocurrencies_au`, `all_cryptocurrencies_ca`, `all_cryptocurrencies_eu`, `all_cryptocurrencies_gb`, and `all_cryptocurrencies_in`.

One of the issues right now though is if you specify a count that's bigger than the number of securities returned from that endpoint, you'll get an error "size is too large".  I'm trying to see if I can find a better way to get that total so the user doesn't have to guess, but right now you have to kind of play around with that `count` argument.

Hope that helps.

**Reply by fja05680** [reply]
Doug, thanks for your quick and helpful response.  Here is some added code that generates the list in case someone else is interested:

```
from yahooquery import Screener

s = Screener()
data = s.get_screeners('all_cryptocurrencies_us', count=250)

# data is in the quotes key
dicts = data['all_cryptocurrencies_us']['quotes']
symbols = [d['symbol'] for d in dicts]
symbols
```
```
Out:
['BTC-USD',
 'ETH-USD',
 'BNB-USD',
 'USDT-USD',
.
.
.
```

**Reply by rfripp2** [reply]
Hi! thanks for the solutions. For me its not working,i get this error:
```
Traceback (most recent call last):
  File "C:\Users\JUAN\Desktop\tradingToolsBackEnd\app.py", line 3, in <module>
    from utils import *
  File "C:\Users\JUAN\Desktop\tradingToolsBackEnd\utils.py", line 162, in <module>
    dicts = data['all_cryptocurrencies_us']['quotes']
KeyError: 'all_cryptocurrencies_us'
```

Any idea why? Thanks!

**Comment by mweber89** [comment]
Was somebody able to pull price, bid and ask from cryptocurrencies?

**Comment by fja05680** [comment]
It would be no different than getting a quote for any security, for example, the symbol for Bitcoin in BTC-USD.  However, the bid/ask doesn't appear on the quote page for Bitcoin:
https://finance.yahoo.com/quote/BTC-USD?p=BTC-USD&.tsrc=fin-srch

That still leaves a question.  How to get real time quotes using yahooquery?  If Bid/Ask available, how to retrieve it?

**Reply by mweber89** [reply]
Exactly. General information looks good. But the quotes are missing..

**Reply by dpguthrie** [reply]
This appears to be specific to cryptocurrencies - the bid, ask, bidSize, and askSize are present when requesting other screeners.  Also, just trying `quotes` does not yield that data either.  Doesn't appear that YF has access or gives access to this data.

**Reply by fja05680** [reply]
BTW - Cryptocurrency info is provided to Yahoo Finance by https://coinmarketcap.com/
So bid/ask may not be available from that source.

Doug, how does one get a real time quote (last price) for a non-crypto symbol like 'SPY'?

**Reply by dpguthrie** [reply]
```
>>> data = Ticker('SPY').quotes
>>> data['SPY']['bid']
392.9
>>> data['SPY']['ask']
392.92
```

**Reply by rfripp2** [reply]
I still get a lot of Coin delisted errors when i get the symbols from CMC and then operate with them with yfinance, and i dont know how to pass the errors and still get the  return for the ones working (besides manually add coin exceptions but thats lot of work and i dont want other person to do it)

**Comment by ersineser** [comment]
How i can retrieve complete list of stock quotes from any country? Example Turkey.
Thanks.

**Reply by ms82494** [reply]
You can accomplish this outside of yahooquery, by creating a screener on the Yahoo Finance website. You'll need a free Yahoo account if you want to save your screener, but you don't need the premium subscription. Just go to the [screener page](https://finance.yahoo.com/screener/), click "Create New Screener", then "Equity Screener". A screener template will pop up. Just remove "United States" from the region, and replace it with Turkey.

**Comment by NeoDad89** [comment]
import yfinance as yf

data = yf.Ticker((list))

print(data.info)(description)

**Comment by algomaschine** [comment]
guys, many thanks! it works awesomely!

**Comment by algomaschine** [comment]
Thanks! I've seen exels in the internet that have tickets, and there are
over hundred thousand tickets. Not sure that screener could handle that...

**Reply by ms82494** [reply]
The web site paginates the results to handle large return sets. @ersineser seemed interested in Turkish stocks, specifically, and the Istanbul exchange lists less than a thousand equities. I've had no problem creating and running the screener for that. For most purposes such a list would not have to be created dynamically every day. I think it would be perfectly reasonable to run the screen on the web site, export the results into a portfolio, and then export that portfolio into a csv file. As a retail investor I wouldn't do this more than once a month.

But if one did need dynamically generated ticker lists, I think one could modify `base.py`, `screener.py`, and `utils/screeners.py` to create a `get_customscreeners` method, analogous to `get_screeners`, that would retrieve the results of custom screens saved under a Yahoo Finance user id.

**Comment by sussyboiiii** [comment]
Hello, 

has someone found out if its possible to get the top 500 or 1000 instead of 250?

Thanks!
